### PR TITLE
feat: added new operators - length_less_than & length_greater_than

### PR DESCRIPTION
**Pull request by kadoshms**
**By submitting this pull request, I confirm that my contribution is made under the terms of the Apache 2.0 license.**

## Description

Added two new operator solvers:

1. `length_less_than`
2. `length_greater_than`

Example usage:

```
metadata:
  id: "test"
definition:
  cond_type: "attribute"
  resource_types:
    - aws_security_group
  attribute: egress.cidr_blocks
  operator: length_less_than
  value: 2
```

Fixes # (issue)

### Description
*Include a description of what makes it a violation and any relevant external links.*

### Fix
*How does someone fix the issue in code and/or in runtime?*

## Checklist:

- [ ] My code follows the style guidelines of this project
- [ ] I have performed a self-review of my own code
- [ ] I have commented my code, particularly in hard-to-understand areas
- [ ] I have made corresponding changes to the documentation
- [ ] I have added tests that prove my feature, policy, or fix is effective and works
- [ ] New and existing tests pass locally with my changes
- [ ] Any dependent changes have been merged and published in downstream modules

### CODE REVIEW

File: checkov/common/checks_infra/solvers/attribute_solvers/length_greater_than_attribute_solver.py [hunk 1]
```diff
@@ -0,0 +1,28 @@
...
+
+
+class LengthGreaterThanAttributeSolver(BaseAttributeSolver):
+    operator = Operators.LENGTH_GREATER_THAN  # noqa: CCE003  # a static attribute
+
+    def _get_operation(self, vertex: Dict[str, Any], attribute: Optional[str]) -> bool:  # type:ignore[override]
+        if vertex.get(attribute) is None:  # type:ignore[arg-type]  # due to attribute can be None
+            return False
+
+        attr = vertex.get(attribute)  # type:ignore[arg-type]  # due to attribute can be None
```
Comment (gruebel) [inline checkov/common/checks_infra/solvers/attribute_solvers/length_greater_than_attribute_solver.py:15]:
```suggestion
        attr = vertex.get(attribute)  # type:ignore[arg-type]  # due to attribute can be None
        if attr is None:  # type:ignore[arg-type]  # due to attribute can be None
            return False
        
```
you can just move it up and then you don't need to call it twice 🙂

File: checkov/common/checks_infra/solvers/attribute_solvers/length_greater_than_or_equal_attribute_solver.py [hunk 2]
```diff
@@ -0,0 +1,11 @@
...
+
+from .length_less_than_attribute_solver import LengthLessThanAttributeSolver
+from checkov.common.graph.checks_infra.enums import Operators
+
+
+class LengthGreaterThanOrEqualAttributeSolver(LengthLessThanAttributeSolver):
+    operator = Operators.LENGTH_GREATER_THAN  # noqa: CCE003  # a static attribute
+
+    def _get_operation(self, vertex: Dict[str, Any], attribute: Optional[str]) -> bool:  # type:ignore[override]
+        return not super()._get_operation(vertex, attribute)
```
Comment (gruebel) [inline checkov/common/checks_infra/solvers/attribute_solvers/length_greater_than_or_equal_attribute_solver.py:11]: sneaky, but smart 😄

File: checkov/common/checks_infra/solvers/attribute_solvers/length_less_than_attribute_solver.py [hunk 3]
```diff
@@ -0,0 +1,28 @@
...
+
+
+class LengthLessThanAttributeSolver(BaseAttributeSolver):
+    operator = Operators.LENGTH_LESS_THAN  # noqa: CCE003  # a static attribute
+
+    def _get_operation(self, vertex: Dict[str, Any], attribute: Optional[str]) -> bool:  # type:ignore[override]
+        if vertex.get(attribute) is None:  # type:ignore[arg-type]  # due to attribute can be None
+            return False
+
+        attr = vertex.get(attribute)  # type:ignore[arg-type]  # due to attribute can be None
```
Comment (gruebel) [inline checkov/common/checks_infra/solvers/attribute_solvers/length_less_than_attribute_solver.py:15]: same here, just move it up and then use `attr`

**Review by gruebel — COMMENTED** [review]
nice job 🍻 please also rename the YAML files, I think you copy pasted it from the `length_equals` solver 😄

**Review by gruebel — APPROVED** [review]
🍻

**Comment by ankon** [comment]
I think `length_less_than_or_equal` doesn't really work properly for the "cop-out" case in the base class: `LengthGreaterThanAttributeSolver._get_operation` returns `True` when the value length cannot be determined, so `LengthLessThanOrEqualAttributeSolver` ends up returning `not True`  in that case -- I would expect it to also return `True`.

Is this intentional?

**Comment by kadoshms** [comment]
@ankon I think you are right there, seems like this "shortcut" wasn't a good cool.
I'll open a fix PR for that as I agree with that.

**Pull request merged**